java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/AdController.java

```diff
@@ -0,0 +1,295 @@
+            if (queue.hasNextAd()) {
+                uiLogsListener.add("Using queue to show the ad");
+                ad = queue.getNextAd();
+                if (ad != null) {
+                    // Be mindful that ads will need to be reattached to your ChartboostMediationFullscreenAdListener if you are listening to ad cycle events.
+                    ad.setListener(createFullscreenAdListener(placementName, correspondingShowButton, uiLogsListener));
+                } else {
+                    uiLogsListener.add("Ad retrieved from the queue is null");
+                }
+                fullscreenAds.put(placementName, ad);
+            } else {
+                uiLogsListener.add("No ads in the queue");
+            }
+        }
+        uiLogsListener.add("Fullscreen ad is about to show");
+        if (ad != null) {
+            ad.showFullscreenAdFromJava(activity, createFullscreenAdShowListener(correspondingShowButton, uiLogsListener));
+        } else {
+            uiLogsListener.add("Fullscreen ad is null. Load an ad first.");
+            correspondingShowButton.setEnabled(false);
+        }
+    }
+
+    final public ChartboostMediationFullscreenAdQueue getAdQueue(String placement) {
+        return adQueues.get(placement);
+    }
+
+    final public void createAdQueue(
+            String placement,
+            Button correspondingShowButton,
+            Context context,
+            UILogsListener uiLogsListener
+    ) {
+        ChartboostMediationFullscreenAdQueue queue = ChartboostMediationFullscreenAdQueueManager.queue(context, placement);
+        queue.setQueueCapacity(UNCONFIGURED_AD_QUEUE_CAPACITY);// this will update queue capacity only to the maximum capacity received on initialization
+        queue.setAdQueueListener(createFullscreenAdQueueListener(correspondingShowButton, uiLogsListener));
+        adQueues.put(placement, queue);
+    }
+
+    final public void setShouldUseQueue(boolean shouldUseQueue) {
+        this.shouldUseQueue = shouldUseQueue;
+    }
+
+    private ChartboostMediationFullscreenAdLoadListener createFullscreenAdLoadListener(
+            String placementName,
+            Button button,
+            UILogsListener uiLogsListener
+    ) {
+        return new ChartboostMediationFullscreenAdLoadListener() {
+            @Override
+            public void onAdLoaded(@NonNull ChartboostMediationFullscreenAdLoadResult chartboostMediationFullscreenAdLoadResult) {
+                if (chartboostMediationFullscreenAdLoadResult.getError() == null) {
+                    uiLogsListener.add("Ad loaded");
+                    fullscreenAds.put(placementName, chartboostMediationFullscreenAdLoadResult.getAd());
+                    button.setEnabled(fullscreenAds.get(placementName) != null);
+                } else {
+                    uiLogsListener.add(String.format("Error loading ad: %s", chartboostMediationFullscreenAdLoadResult.getError().getMessage()));
```

Comment:
Here and elsewhere, I'd also get and log the error code. That allows for quicker lookups.